roadmap/implementors-guide/guide.md

```diff
@@ -475,9 +475,12 @@ Utility structs:
 /// Metadata used to track previous parachain validation code that we keep in
 /// the state.
 pub struct ParaPastCodeMeta {
-	// Block numbers where the code was replaced. These can be used as indices
+	// Block numbers where the code was expected to be replaced and where the code
+	// was actually replaced, respectively. The first is used to do accurate lookups
+	// of historic code in historic contexts, whereas the second is used to do
+	// pruning on an accurate timeframe. These can be used as indices
 	// into the `PastCode` map along with the `ParaId` to fetch the code itself.
-	upgrade_times: Vec<BlockNumber>,
+	upgrade_times: Vec<(BlockNumber, BlockNumber)>,
```

Comment:
@burdges I also changed things here to have two indices - the `expected_at` and the actual block number of inclusion.